pkg/apis/componentconfig/types.go

```diff
@@ -31,9 +31,13 @@ import (
 type ControllerManagerConfiguration struct {
 	// Address is the IP address to serve on (set to 0.0.0.0 for all interfaces).
 	Address string
-	// Port is the port that the controller's http service runs on.
+
+	// DEPRECATED/Ignored, use SecurePort instead.
 	Port int32
 
+	// SecurePort is the port that the controller's https service runs on.
```

Comment:
I think normally we just embed `SecureServingOptions` into the configuration struct. We might also want to add the `InsecureServingOptions` if people want a way to serve metrics insecurely.

Reply:
Perhaps I'm missing your point here, but simply embedding SecureServingOptions in our config struct doesn't buy us anything.  We aren't making use of the k8s.io/apiserver package in the controller outside of the additions I'm doing in this PR to , so I must manually invoke MaybeDefaultWithSelfSignedCerts() to setup the cert and key files.  If the only change you are looking for is to add SecureServingOption to the struct instead of it being a local var I can do that but I think I'm missing your point.